src/Context/ContextFactory.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Roave\BetterReflection\Context;
+
+interface ContextFactory
+{
+    /**
+     * Build a Context for a namespace in the provided file contents.
+     *
+     * @param string $namespace It does not matter if a `\` precedes the namespace name, this method first normalizes.
```

Comment:
This is something that should be enforced by a value object, not by a docblock comment and the implementation